src/mesh/tdycoremesh.c

```diff
@@ -387,110 +361,120 @@ PetscErrorCode SaveMeshGeometricAttributes(TDy tdy) {
   TDyVertex     *vertices = &mesh->vertices;
   TDyEdge       *edges = &mesh->edges;
   TDyFace       *faces = &mesh->faces;
-  PetscInt       dim;
-  PetscInt       cStart, cEnd;
-  PetscInt       vStart, vEnd;
-  PetscInt       eStart, eEnd;
-  PetscInt       fStart, fEnd;
-  PetscInt       pStart, pEnd;
-  PetscInt       icell, iedge, ivertex, ielement, iface, d;
-  PetscErrorCode ierr;
 
-  ierr = DMGetDimension(dm, &dim); CHKERRQ(ierr);
+  // Determine the number of cells, edges, and vertices of the mesh
+  PetscInt c_start, c_end;
+  PetscErrorCode ierr = DMPlexGetHeightStratum(dm, 0, &c_start, &c_end); CHKERRQ(ierr);
```

Comment:
Because `ierr` is used in subsequent lines, how about instead using the following?
```
PetscErrorCode ierr;
ierr = DMPlexGetHeightStratum(dm, 0, &c_start, &c_end); CHKERRQ(ierr); 
```